### PR TITLE
CIF data

### PR DESCRIPTION
**Pull request by kjappelbaum**
for a subset 
Screenshot 2023-10-22 at 21 09 23

introduces a hack in `text_sampling.py` to deal with empty strings 

searched for cases in which we have `SMILES {SMILES#}`

### CODE REVIEW

File: data/tabular/mp_self_supervised/prepare_data.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "spacegroup_number": sg_n,
+        "cif": cif,
+        "is_longer_than_allowed": is_longer_than_allowed(cif),
+    }
+    return d
+
+
+if __name__ == "__main__":
+    all_structures = glob(
+        "/Users/kevinmaikjablonka/LSMO Dropbox/Kevin Jablonka/small_tests/download_mp/structures/optimade_materialsproject_org/*.cif"
```
Comment (kjappelbaum) [inline data/tabular/mp_self_supervised/prepare_data.py:30]:
```suggestion
        "structures/*.cif" #assumes structures have been downloaded
 ```

File: data/tabular/qmof_gcmc/transform.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import ast
+
+import pandas as pd
+import numpy as np
+
+
+def process():
+    df = pd.read_json(
+        "/Users/kevinmaikjablonka/LSMO Dropbox/Kevin Jablonka/mofdscribe_data/qmof/v0.0.1/data.json"  # "https://huggingface.co/datasets/kjappelbaum/chemnlp-qmof-data/resolve/main/qmof_data.json"
```
Comment (kjappelbaum) [inline data/tabular/qmof_gcmc/transform.py:9]:
```suggestion
        "https://huggingface.co/datasets/kjappelbaum/chemnlp-qmof-data/resolve/main/qmof_data.json"
```

**Review by MicPie — APPROVED** [review]
lgtm

File: data/tabular/block_polymers_morphology/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          - noun: phase
+          - noun: experimentally observed phase
+    - id: T
+      description: temperature of measurement
+      type: continuous
+      significant_digits: 0
+      units: K
+    - id: Mn
+      description: number-average molar mass
+      type: continuos
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply data/tabular/block_polymers_morphology/meta.yaml:19]: oh! And I always tell people to use spellcheckers. Embarrassing!

**Pull request merged**